### PR TITLE
Update version ingestion to pull from package data

### PR DESCRIPTION
**Pull request by samuelli**
Part of #531 

Follows from #984

### CODE REVIEW

File: src/manage.py [hunk 1]
```diff
@@ -217,8 +217,7 @@ def pull_registry_info(self):
     # Check if there's a NPM scope or the default scope @@npm.
     assert self.scope.startswith('@')
 
-    headers = {'Accept': 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*'}
-    response = urlfetch.fetch(util.npm_registry_url(self.scope, self.package), headers=headers, validate_certificate=True)
+    response = util.registry_get(self.scope, self.package)
```
Comment (dstockwell) [inline src/manage.py]: this will need to deal with failure, maybe throw an exception before self.retry?

File: src/manage.py [hunk 2]
```diff
@@ -230,6 +229,14 @@ def pull_registry_info(self):
 
       if self.owner == '' or self.repo == '':
         self.error('No github URL associated with package', ErrorCodes.Library_no_github)
+
+        new_metadata = json.loads(response.content)
```
Comment (dstockwell) [inline src/manage.py]: is this meant to be inside the error case?

File: src/manage.py (hunk 1 above)
Comment (dstockwell) [inline src/manage.py]: is there an etag? how can we avoid fetching this every update, maybe it doesn't matter since there's no quota...

Comment (samuelli) [inline src/manage.py]: There's no etag... it's really quite a terrible, undocumented API. There's no quota, rate limits so I think its ok. I believe the `_rev` will let us know if it has changed so at least we can guard processing/database hits.

File: src/manage.py (hunk 2 above)
Comment (samuelli) [inline src/manage.py]: No. I'm guessing you wrote this yesterday, its been fixed since then.

File: src/manage.py (hunk 1 above)
Comment (samuelli) [inline src/manage.py]: Not sure what you mean? The response status code is checked below

Comment (dstockwell) [inline src/manage.py]: Oh, yeah.

File: src/manage.py [hunk 3]
```diff
@@ -217,8 +217,7 @@ def update_registry_info(self):
     # Check if there's a NPM scope or the default scope @@npm.
     assert self.scope.startswith('@')
 
-    headers = {'Accept': 'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*'}
-    response = urlfetch.fetch(util.npm_registry_url(self.scope, self.package), headers=headers, validate_certificate=True)
+    response = util.registry_get(self.scope, self.package)
     try:
       package = json.loads(response.content)
```
Comment (dstockwell) [inline src/manage.py]: shouldn't this be inside the `status_code == 200` case?

Comment (samuelli) [inline src/manage.py]: Hmm, those lines aren't really needed. I think we want to assume that we are getting a JSON response and fail loudly if the registry isn't complying with that.

Comment (samuelli) [inline src/manage.py]: Removed

**Pull request merged**